### PR TITLE
Uni 55658 create standalone demo of second hot reload crash

### PR DESCRIPTION
**Pull request by lassond**
### What does this implement/fix? Explain your changes.
This adds a standalone application that reproduce the thrown exception on second domain unload (System.AppDomainUnloadedException: Attempted to access an unloaded AppDomain)

### Does this close any currently open issues?

...

### Any other comments?

...

### Checklist

Check all those that are applicable and complete.

-   [ ] Make sure to include one or more tests for your change
-   [ ] If an enhancement PR, please create docs and at best an example
-   [ ] Add yourself to [`AUTHORS`](../blob/master/AUTHORS.md)
-   [ ] Updated the [`CHANGELOG`](../blob/master/CHANGELOG.md)

### CODE REVIEW

File: hotReloadCrashRepro/Program.cs [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+    {
+        /// <summary>
+        /// Args goes as follows:
+        ///     0: The full path to theAssembly.cs 
+        /// </summary>
+        /// <param name="args"></param>
+        static void Main(string[] args)
+        {
+            // Defaults is missing args
+            string pathToTheAssembly = @"D:\projects\pythonnet\hotReloadCrashRepro\theAssembly.cs";
```
Comment (benoithudson) [inline hotReloadCrashRepro/Program.cs]: Can you use a relative path?

File: hotReloadCrashRepro/Program.cs [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+            if (args.Length > 0)
+            {
+                pathToTheAssembly = args[0];
+            }
+
+            // The exception is thrown on the second call to Py_Finalize
```
Comment (benoithudson) [inline hotReloadCrashRepro/Program.cs:38]: Needs more doc: at a high level, why is (or, soon, was) this causing a problem?

Comment (benoithudson) [reply hotReloadCrashRepro/Program.cs:38]: (Namely: first time through, on the python side we litter some objects that python figures someone still has a reference to, so it keeps them around -- leak! Second time through, python gc looks at the leaked objects and calls tp_traverse on them. But the tp_traverse handler is C# code that got destroyed in the domain unload -- crash!)

File: hotReloadCrashRepro/Properties/AssemblyInfo.cs [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+// General Information about an assembly is controlled through the following
+// set of attributes. Change these attribute values to modify the information
+// associated with an assembly.
+[assembly: AssemblyTitle("hotReloadCrashRepro")]
+[assembly: AssemblyDescription("")]
+[assembly: AssemblyConfiguration("")]
+[assembly: AssemblyCompany("HP Inc.")]
+[assembly: AssemblyProduct("hotReloadCrashRepro")]
+[assembly: AssemblyCopyright("Copyright © HP Inc. 2018")]
```
Comment (benoithudson) [inline hotReloadCrashRepro/Properties/AssemblyInfo.cs:13]: HP?

**Review by benoithudson — CHANGES_REQUESTED** [review]
More doc please, and relative paths if possible.

File: hotReloadCrashRepro/Properties/AssemblyInfo.cs (hunk 3 above)
Comment (lassond) [reply hotReloadCrashRepro/Properties/AssemblyInfo.cs:13]: This is the file that comes with a default new .NET console application, using Visual Studio 2017. I will not change it unless you insist

**Comment by lassond** [comment]
@benoithudson : I pushed the requested modifications, can you take a look please?

**Review by benoithudson — APPROVED** [review]

**Pull request merged**